src/PostProcesses/postProcess.ts

```diff
@@ -677,17 +732,32 @@ export class PostProcess {
 
     private _disposeTextures() {
         if (this._shareOutputWithPostProcess || this._forcedOutputTexture) {
+            this._disposeTextureCache();
             return;
         }
 
         if (this._textures.length > 0) {
             for (var i = 0; i < this._textures.length; i++) {
                 this._engine._releaseTexture(this._textures.data[i]);
+
+                const cacheIndex = this._textureCache.indexOf(this._textures.data[i]);
+                if (cacheIndex !== -1) {
+                    this._textureCache.splice(cacheIndex, 1);
+                }
             }
         }
+
+        this._disposeTextureCache();
         this._textures.dispose();
     }
 
+    private _disposeTextureCache() {
+        for (let i = this._textureCache.length - 1; i >= 0; i--) {
+            this._engine._releaseTexture(this._textureCache[i]);
+            this._textureCache.splice(i, 1);

```

Comment:
nit: you could remove this line and add `this._textureCache.length = 0;` after the loop